src/classes/UTIL_JobProgress_CTRL.cls

```diff
@@ -31,139 +31,163 @@
                                     CreatedDate, CreatedById, CompletedDate, ApexClassId, ApexClass.Name, CreatedBy.Name
-                                    FROM AsyncApexJob WHERE JobType='BatchApex' ORDER BY CreatedDate DESC, Id LIMIT :numberOfJobs]) {
-            Double itemsProcessed = a.JobItemsProcessed;
-            Double totalItems = a.TotalJobItems;
-
-            BatchJob j = new BatchJob();
-            j.job = a;
-
-            j.percentComplete = 5;
-            //Determine the pecent complete based on the number of batches complete
-            if(a.status == 'Completed') {
-                j.percentComplete = 100;
-            //Determine the pecent complete based on the number of batches complete
-            } else if (a.status == 'Queued') {
-                j.percentComplete = 5;
-            } else if (a.status == 'Preparing') {
-                j.percentComplete = 10;
-            } else if (totalItems == 0) {
-                //A little check here as we don't want to divide by 0.
-                j.percentComplete = 5;
-            }
-            else {
-                j.percentComplete = ((itemsProcessed  / totalItems) * 100.0).intValue();
-            }
+                                    FROM AsyncApexJob
+                                    WHERE JobType='BatchApex'
+                                    AND (ApexClass.Name != :skewDispatcherClassName
+                                        OR (ApexClass.Name = :skewDispatcherClassName AND Status = 'Holding'))
+                                    AND CreatedDate >= :batchJobStartDT
+                                    AND CreatedById = :currUserId
+                                    ORDER BY CreatedDate DESC, Id LIMIT :numberOfJobs]) {
 
-            //don't let it go backwards from preparing/queued
-            if (j.percentComplete == 0)
-                j.percentComplete = 5;
+            BatchJob j = new BatchJob(a);
 
-            if (a.ExtendedStatus != null && a.ExtendedStatus != '')
-                hasExtendedStatus = true;
+            if (!String.isEmpty(a.ExtendedStatus)) {
+                this.hasExtendedStatus = true;
+            }
 
-            if (a.Status == 'Completed'){
+            if (a.Status == 'Completed') {
                 completedCount++; 
             }
 
             batchJobs.add(j);
         }
-        if (completedCount == numberOfJobs)
-            allCompleted = true;
-        if (!batchJobs.isEmpty())
-            bj = batchJobs[0];
+        if (completedCount == this.numberOfJobs) {
+            this.allCompleted = true;
+        }
+        if (!batchJobs.isEmpty()) {
+            this.bj = batchJobs[0];
+        }
+
         return batchJobs;
     }
 
     //This is the wrapper class the includes the job itself and a value for the percent complete
-    public Class BatchJob{
+    public class BatchJob {
```

Comment:
@force2b this wrapper class BatchJob and the wrapper class BatchJobStatus look almost the same. Should we merge them?